Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/sites/chan55/Chan55.java

```diff
@@ -0,0 +1,308 @@
+import com.github.adamantcheese.chan.core.site.http.Reply;
+import com.github.adamantcheese.chan.core.site.http.ReplyResponse;
+import com.github.adamantcheese.chan.core.site.sites.dvach.Dvach;
+import com.github.adamantcheese.chan.core.site.sites.dvach.DvachReplyCall;
+import com.github.adamantcheese.chan.ui.layout.ReplyLayout;
+import com.github.adamantcheese.chan.utils.Logger;
+
+import org.jsoup.Jsoup;
+
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import okhttp3.HttpUrl;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class Chan55 extends CommonSite {
+    public static final CommonSiteUrlHandler URL_HANDLER = new CommonSiteUrlHandler() {
+        @Override
+        public Class<? extends Site> getSiteClass() {
+            return Chan55.class;
+        }
+
+        @Override
+        public HttpUrl getUrl() {
+            return HttpUrl.parse("https://55chan.org/");
+        }
+
+        @Override
+        public String[] getNames() {
+            return new String[]{"55chan"};
+        }
+
+        @Override
+        public String desktopUrl(Loadable loadable, @Nullable Post post) {
+            if (loadable.isCatalogMode()) {
+                return getUrl().newBuilder().addPathSegment(loadable.boardCode).toString();
+            } else if (loadable.isThreadMode()) {
+                return getUrl().newBuilder()
+                        .addPathSegment(loadable.boardCode).addPathSegment("res")
+                        .addPathSegment(loadable.no + ".html")
+                        .toString();
+            } else {
+                return getUrl().toString();
+            }
+        }
+    };
+
+    private static final String TAG = "Chan55";
+    private static final String BASE_URL = "https://55chan.org";
+    private final Random secureRandom = new SecureRandom();
+
+    @Override
+    public void setup() {
+        setName("55chan");
+        setIcon(SiteIcon.fromFavicon(HttpUrl.parse("https://55chan.org/static/favicon.ico?v=2")));
+        setBoardsType(BoardsType.STATIC);
+
+        setResolvable(URL_HANDLER);
+
+        setConfig(new CommonConfig() {
+            @Override
+            public boolean feature(Feature feature) {
+                return feature == Feature.POSTING || feature == Feature.POST_DELETE;
+            }
+        });
+
+        setEndpoints(new VichanEndpoints(this,
+                BASE_URL,
+                BASE_URL) {
+            @Override
+            public HttpUrl imageUrl(Post.Builder post, Map<String, String> arg) {
+                return root.builder().s(post.board.code).s("src").s(arg.get("tim") + "." + arg.get("ext")).url();
+            }
+
+            @Override
+            public HttpUrl thumbnailUrl(Post.Builder post, boolean spoiler, Map<String, String> arg) {
+                String ext;
```

Comment:
You're doing something weird here. Set `ext` to be `arg.get("ext")` to start and switch off the variable. If it's any of the cases, just break, and in the default case reset it. No need to be calling the function twice.